hoomd/simulation.py

```diff
@@ -213,7 +214,10 @@ def create_state_from_snapshot(self, snapshot):
     @property
     def state(self):
         """hoomd.State: The current simulation state."""
-        return self._state
+        if self._state is None:
+            raise AttachedDataError("state")
```

Comment:
This is a different sort of error than the others. Internally the state can exist without operations being attached.

Reply:
This is true. Shall I just add another type of error here or just pass `None`?

Reply:
I think we should just return `None` here. `None` is a clear indication that the state does not yet exist because it has not been created.

Reply:
I think `None` is the correct solution here; we are making a statement that the state is null here.